src/execution/index/art/node.cpp

```diff
@@ -186,4 +188,208 @@ void Node::Erase(Node *&node, idx_t pos, ART &art) {
+
+	if (!l_node) {
+		l_node = r_node;
+		if (l_node_parent) {
+			l_node_parent->ReplaceChildPointer(l_node_pos, l_node);
+		}
+		if (r_node_parent) {
+			r_node_parent->ReplaceChildPointer(r_node_pos, nullptr);
+		}
+		r_node = nullptr;
+		return;
+	}
+
+	// make sure that r_node has the longer (or equally long) prefix
+	if (l_node->prefix.Size() > r_node->prefix.Size()) {
+
+		auto temp_art = l_art;
+		l_art = r_art;
+		r_art = temp_art;
+
+		auto temp_node = l_node;
+		l_node = r_node;
+		r_node = temp_node;
+
+		if (l_node_parent) {
+			l_node_parent->ReplaceChildPointer(l_node_pos, l_node);
+		}
+		if (r_node_parent) {
+			r_node_parent->ReplaceChildPointer(r_node_pos, r_node);
+		}
+	}
+
+	// get first mismatch position
+	auto mismatch_pos = l_node->prefix.Size();
+	for (idx_t i = 0; i < l_node->prefix.Size(); i++) {
+		if (l_node->prefix[i] != r_node->prefix[i]) {
+			mismatch_pos = i;
+			break;
+		}
+	}
+
+	// no prefix or same prefix
+	if (mismatch_pos == l_node->prefix.Size() && l_node->prefix.Size() == r_node->prefix.Size()) {
+		return Node::Merge(l_art, r_art, l_node, r_node, depth + mismatch_pos, l_node_parent, l_node_pos, r_node_parent,
+		                   r_node_pos);
+	}
+
+	if (mismatch_pos == l_node->prefix.Size()) {
+		// r_node's prefix contains l_node's prefix
+		// l_node cannot be a leaf, otherwise the key represented by l_node would be a subset of another key
+		// which is not possible by our construction
+		D_ASSERT(l_node->type != NodeType::NLeaf);
+
+		// test if the next byte (mismatch_pos) in r_node (longer prefix) exists in l_node
+		auto mismatch_byte = r_node->prefix[mismatch_pos];
+		auto child_pos = l_node->GetChildPos(mismatch_byte);
+
+		// update the prefix of r_node to only consist of the bytes after mismatch_pos
+		r_node->prefix.Reduce(mismatch_pos);
+
+		if (child_pos == DConstants::INVALID_INDEX) {
+			Node::InsertChildNode(l_node, mismatch_byte, r_node);
+			if (l_node_parent) {
+				l_node_parent->ReplaceChildPointer(l_node_pos, l_node);
+			}
+			if (r_node_parent) {
+				r_node_parent->ReplaceChildPointer(r_node_pos, nullptr);
+			}
+			r_node = nullptr;
+			return;
+		}
+
+		auto child_node = l_node->GetChild(*l_art, child_pos);
+		return Node::ResolvePrefixesAndMerge(l_art, r_art, child_node, r_node, depth + mismatch_pos, l_node, child_pos,
+		                                     r_node_parent, r_node_pos);
+	}
+
+	// prefixes differ, create new node and insert both nodes as children
+
+	// create new node
+	Node *new_node = new Node4();
+	new_node->prefix = Prefix(l_node->prefix, mismatch_pos);
+
+	// insert l_node, break up prefix of l_node
+	auto key_byte = l_node->prefix.Reduce(mismatch_pos);
+	Node4::Insert(new_node, key_byte, l_node);
+
+	// insert r_node, break up prefix of r_node
+	key_byte = r_node->prefix.Reduce(mismatch_pos);
+	Node4::Insert(new_node, key_byte, r_node);
```

Comment:
I think this code is rather similar to the one in the `Insert` can we use that here?